document/chapters/motivation.tex

```diff
@@ -1,3 +1,73 @@
+
+This limitations create some ergonomic problems with arrays: Two arrays with different sizes have different types, even if both store values of the same type. Thus, trait implementations for array types must be done manually for each possible size. Because of this, the standard library only implements traits for array types up to a size of 32, even though arrays are a primitive type of the language. In certain cases vectors are used instead of arrays. Making the code unsuitable for applications running on embedded devices (which may not allow heap allocation) or applications needing multidimensional vectors.
+
+An example of this limitation can be found on \ref{lst:trait_array}, where the implementation of the trait \texttt{Volatile} cannot be easily extended to other array sizes, even when such implementation does not use the size of the array. 
+
+\begin{listing}
+	\begin{minted}{rust} 
+    const N: usize = 3;
+
+    trait Volatile {
+        fn explode(&self);
+    }
+
+    impl<T> Volatile for [T; N] {
+        fn explode(&self) {
+            for _ in self {
+                println!("Boom!")
+            }
+        }
+    }
+
+    fn main() {
+        [0i32, 1, 2].explode(); 
+        [0i32, 1].explode(); // stderr: no method named `explode` 
+                             // found for type `[i32; 2]` in the
+                             // current scope
+    }
+	\end{minted}
+    \caption{Even though \texttt{Volatile} is implemented for \texttt{[T; 3]}, it is not for \texttt{[T;2]}.}
+  \label{lst:trait_array}
+\end{listing}
+
+\section{Static control flow and optimizations}
+Rust's compiler is capable of doing static control flow to improve performance execution. If a control flow statement can be resolved during compilation, the compiler will remove the unused expressions of the statement. For example, if the condition of an if-else statement can be resolved during compilation, the compiler will replace the statement with one just including the matching arm of the statement.
+
+An small illustration of this can be seen on \ref{lst:static_control_flow}, where the array size \texttt{N} is a constant value known during compilation. Given that the expression  \texttt{N > 0} can be evaluated during compilation, the compiler will optimize the function, resulting in code similar to \ref{lst:optimized}. However, doing such optimizations increases compilation time. 
```

Comment:
```suggestion
An small illustration of this can be seen in Listing \ref{lst:static_control_flow}, where the array size \texttt{N} is a constant value known during compilation. Given that the expression  \texttt{N > 0} can be evaluated during compilation, the compiler will optimize the function, resulting in code similar to the code shown in Listing \ref{lst:optimized}. However, doing such optimizations increases compilation time. 
```